src/store/participantsStore.spec.js

```diff
@@ -572,27 +572,109 @@ describe('participantsStore', () => {
 			})
 		})
 
-		test('joins call', async () => {
-			// Assert
+		afterEach(() => {
+			jest.clearAllTimers()
+		})
+
+		const assertInitialCallState = () => {
 			expect(joinCall).toHaveBeenCalledWith(TOKEN, flags, false, false)
+			EventBus.emit('signaling-join-call', [TOKEN, actualFlags])
 			expect(store.getters.isInCall(TOKEN)).toBe(true)
 			expect(store.getters.isConnecting(TOKEN)).toBe(true)
-			expect(store.getters.participantsList(TOKEN)).toStrictEqual([
-				{
-					attendeeId: 1,
-					sessionId: 'session-id-1',
-					inCall: actualFlags,
-					participantType: PARTICIPANT.TYPE.USER,
-				},
-			])
+			expect(store.getters.isJoiningCall(TOKEN)).toBe(false)
+			expect(store.getters.participantsList(TOKEN)).toStrictEqual([{
+				attendeeId: 1,
+				sessionId: 'session-id-1',
+				inCall: actualFlags,
+				participantType: PARTICIPANT.TYPE.USER,
+			}])
+		}
 
-			// Finished connecting to the call
-			EventBus.emit('signaling-users-in-room')
+		const finishConnectingWithUsers = (participants) => {
```

Comment:
make the same as `assertParticipantsListReceivedBeforeJoiningCall = (event, payload)` to test also `'joins call with external signaling'`?